elide-datastore/elide-datastore-jpa/src/main/java/com/yahoo/elide/datastores/jpa/SupplierEntityManager.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2023, the original author or authors.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.jpa;
+
+import jakarta.persistence.EntityGraph;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.FlushModeType;
+import jakarta.persistence.LockModeType;
+import jakarta.persistence.Query;
+import jakarta.persistence.StoredProcedureQuery;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaDelete;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.CriteriaUpdate;
+import jakarta.persistence.metamodel.Metamodel;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * EntityManager used to proxy to a target entity manager.
+ */
+public class SupplierEntityManager implements EntityManager {
```

Comment:
I think the names for SupplierEntityManager and EntityManagerSupplier clash.   Not sure what would be better though.

Reply:
Yeah this really isn't ideal but I think `SupplierEntityManager` reflects what it does. Spring has got quite a number of implementations like `SupplierCsrfToken` and `SupplierJwtDecoder` that does the same thing and delegates to the underlying supplier. One option was to have named this `EntityManagerProxy` which would better describe it's intended use but Spring already had a similarly named interface. Would this be better? 

The other option would be renaming `EntityManagerSupplier` which is hard too. It's really more of a factory but `EntityManagerFactory` would clash too.